### DISCUSSION TITLE
Not ask the user if the app supports TypeScript

### DISCUSSION THREAD

**Discussion by smolinari** [quasarframework/app-extension-apollo#107] [General]
@ejez @IlCallo 

Question: Couldn't we automate the question of if the app support TS or not during installation of the AE? By that I mean, not ask the question to the user at all, but rather introspect the app for signs it is working with TS? It should be possible, right? Shouldn't there even be a flag set in quasar.conf.js?

Scott

**Comment by IlCallo** [comment]
We are able to do it, we just need to check if there's a tsconfig at root level.
But last time we discussed this on Discord IIRC we couldn't come up with a common ground different than asking the user to opt-in, and the whole Quasar ecosystem is currently using it like this.

We probably will need to reconsider it when we start working again on generating TS files for all modes in Q3

**Comment by yusufkandemir** [comment]
Freshly introduced `api.hasTypescript()` API method makes this easily possible: https://github.com/quasarframework/quasar/blob/abd5620d4d3b2b6c1d035d727ce853c1e6c70a16/app-vite/types/app-extension.d.ts#L36
We need to enforce newer versions of `@quasar/app-vite` and `@quasar/app-webpack`. Here is an example:
https://github.com/quasarframework/quasar-testing/blob/75e4fb524fd7767492a1cac66fe8169a5b29b6a6/packages/e2e-cypress/src/install.js#L67-L76

We will be dropping support for the outdated `@quasar/app` in #141, which can be followed by bumping the lowest supported version even more. I would advise making a release with that PR first so users can start using that feature without having to upgrade their app-* versions to the latest. We can ship this automatic TS detection in another release.

I will create a PR for this after #141 gets merged.

**Reply by yusufkandemir** [reply]
I've created the PR for it: https://github.com/quasarframework/app-extension-apollo/pull/148

**Reply by yusufkandemir** [reply]
Available since [2.2.0-beta.1](https://github.com/quasarframework/app-extension-apollo/releases/tag/2.2.0-beta.1)
